### ISSUE TITLE
"Unexpected token :" for JSON-LD (ld+json).

### ISSUE THREAD

**Issue by jaepage** [SublimeLinter/SublimeLinter-eslint#194]
If I place a JSON-LD script element in an HTML file, ESLint throws an error of "Unexpected token :". 

Google doc: https://developers.google.com/search/docs/guides/intro-structured-data

        <script type="application/ld+json">
            {
                "@context": "http://schema.org",
                "@type": "Organization",
                "url": "http://www.example.com",
                "name": "Unlimited Ball Bearings Corp.",
                "contactPoint": {
                    "@type": "ContactPoint",
                    "telephone": "+1-401-555-1212",
                    "contactType": "Customer service"
                }
            }
        </script>

Disabling ESLint in Sublime Text menubar's "Toggle" command removes the error. 

Surrounding the entire JSON object with `/* eslint-disable */` and `/* eslint-enable */` doesn't remove the error. 

Should this be reported here or at [eslint](https://github.com/eslint/eslint/issues)? 

Can this be prevented, please? 

- ESLint 4.17.0
- SublimeLinter 3.10.10
- Sublime Text 3.0
- macOS 10.13.3 High Sierra

**Comment by kaste** [comment]
Hm, what does it tell you if you lint on the command line?

**Comment by jaepage** [comment]
```
  1:2  error  Parsing error: Unexpected token !

✖ 1 problem (1 error, 0 warnings)
```

**Comment by kaste** [comment]
On line 1? Do you use the eslint html plugin?

**Comment by jaepage** [comment]
Do I use the ESLint HTML plugin on line one… I apologize, I don't understand your question. Line one of my HTML is `<!DOCTYPE html>`.

**Comment by kaste** [comment]
Haha. How do you tell eslint on the command line to lint a html file?

**Comment by jaepage** [comment]
I don't usually lint on the command line. I prefer the highlighting of code as I write in Sublime. 

When you asked me to lint on the command line above, I ran `eslint index.html`.

**Comment by kaste** [comment]
Is that even valid js inside the script tag? I mean it's json, okay.

But Sublime itself might report this as js. Maybe you can check what scope Sublime returns here. 

And you can paste the content of the script, in a new view, and see if eslint would accept this as valid JavaScript.

**Comment by jaepage** [comment]
The script element quoted above is directly pasted from the linked Google doc. Its type is JSON-LD, not JavaScript, so I wouldn't expect ESLint even to be inspecting it. 

How would I "check what scope Sublime returns here" (I don't understand what you're asking)?

**Comment by kaste** [comment]
There is a plugin scope hunter, which shows you the syntax scope the cursor is on. 

I guess Sublime thinks this is JavaScript, and thus eslint runs.

**Comment by jaepage** [comment]
If Sublime or SublimeLinter thinks this script element is JS, despite the type being JSON-LD, how might I correct that?

**Comment by kaste** [comment]
**if** sublime's syntax file here is the reason, then you just have to exclude that whole file for eslint. A PR against Sublime would be hard.

**Comment by jaepage** [comment]
Yes, the only workaround I've found is to exclude the file from ESLint, which isn't ideal. 

I'm surprised that surrounding the JSON with disable/enable comments doesn't work.

**Comment by kaste** [comment]
#188 could solve the issue probably, but you would have to install the eslint html plugin.

**Comment by jaepage** [comment]
In exploring #188, I installed `eslint-plugin-html`, and added the minimally required ` "plugins": ["html"]` to my `.eslintrc`. But it made no difference, nor did adding `"settings": { "html/javascript-mime-types": ["text/javascript", "text/jsx"] }`, which I hoped would exclude JSON mime types.

**Comment by kaste** [comment]
You have to check this on the command line. If #188 lands this would then also work within SublimeLinter.

**Comment by jaepage** [comment]
If I: 
- re-install `eslint-plugin-html`, 
- add the `"plugins"` config code above, 
- add or omit the `"settings"` config code above, 
- then run `eslint index.html`… 

…it produces no output on the command line. 

If I introduce an obvious error to a different script element in that HTML file, it continues to produce no input. 

It appears this prevents ESLint from doing anything with HTML from the command line (though it continues to operate in Sublime). So I'm not sure where we're going with that.

**Comment by kaste** [comment]
eslint and its plugins must be installed in the same scope. So, a locally installed eslint will only use locally installed plugins. 

Actually, it works pretty well. I use it with 

```
<!DOCTYPE html>
<html>
<head>
  <title></title>
</head>
<body>


  <script type="text/javascript">

    Hi I'm broken
  </script>

  <script type="application/ld+json">
          {sss
              "@context": "http://schema.org",
              "@type": "Organization",
              "url": "http://www.example.com",
              "name": "Unlimited Ball Bearings Corp.",
              "contactPoint": {
                  "@type": "ContactPoint",
                  "telephone": "+1-401-555-1212",
                  "contactType": "Customer service"
              }
          }
  </script>

</body>
</html>
```

and get 

```
> eslint index.html

D:\test-eslint-html-plugin\index.html
  11:8  error  Parsing error: Unexpected token I

✖ 1 problem (1 error, 0 warnings)


```

So the json stuff gets ignored. (I intentionally broke the json in my index.html)

**Comment by kaste** [comment]
Closing now, bc we already intent to support the eslint-html-plugin be it via #188 or a different solution.

**Comment by jaepage** [comment]
> eslint and its plugins must be installed in the same scope. So, a locally installed eslint will only use locally installed plugins.

I installed both `eslint` and `eslint-plugin-html` globally.

**Comment by jaepage** [comment]
Would the eventual solution work also in Sublime app, not just the command line?

**Comment by kaste** [comment]
Yes.

**Comment by p-rk** [comment]
You can set for dangerously 

` const json = `{
            "@context": "http://schema.org",
            "@type": "Organization",
            "url": "http://www.example.com",
            "name": "Unlimited Ball Bearings Corp.",
            "contactPoint": {
                "@type": "ContactPoint",
                "telephone": "+1-401-555-1212",
                "contactType": "Customer service"
            }
        }``;

In your HTML

```
          <script
            type="application/ld+json"
            dangerouslySetInnerHTML={{ __html: json }}
          />
```
